pkg/blob/controllerserver.go

```diff
@@ -667,6 +681,94 @@ func (d *Driver) DeleteBlobContainer(ctx context.Context, subsID, resourceGroupN
 	})
 }
 
+// CopyBlobContainer copies a blob container in the same storage account
+func (d *Driver) CopyBlobContainer(ctx context.Context, req *csi.CreateVolumeRequest, accountKey, dstContainerName, storageEndpointSuffix string, secrets map[string]string) error {
+	resourceGroupName, accountName, srcContainerName, _, subsID, err := GetContainerInfo(req.GetVolumeContentSource().GetVolume().GetVolumeId())
+	if err != nil {
+		return status.Error(codes.NotFound, err.Error())
+	}
+	klog.V(2).Infof("begin to copy blob container %s to %s", srcContainerName, dstContainerName)
+	if srcContainerName == "" {
+		return fmt.Errorf("srcContainerName is empty")
+	}
+	if dstContainerName == "" {
+		return fmt.Errorf("dstContainerName is empty")
+	}
+	return wait.ExponentialBackoff(d.cloud.RequestBackoff(), func() (bool, error) {
+		var copyErr error
+		var out []byte
+		if len(secrets) > 0 {
+			accountSasToken, err := getAccountSasToken(secrets, accountName)
+			if err != nil || accountSasToken == "" {
+				klog.V(2).Infof("get account sas token from secrets err: %s, generate sas token for account(%s)", err.Error(), accountName)
+				accountSasToken, err = GenerateSASToken(accountName, accountKey, storageEndpointSuffix)
+				if err != nil {
+					return true, err
+				}
+			}
+			srcContainer, getErr := getContainerReference(srcContainerName, secrets, d.cloud.Environment)
+			if getErr != nil {
+				return true, getErr
+			}
+			exists, err := srcContainer.Exists()
+			if err != nil {
+				return true, err
+			}
+			if !exists {
+				return true, fmt.Errorf("source container(%s) does not exist", srcContainerName)
+			}
+			srcPath := fmt.Sprintf("https://%s.blob.%s/%s%s", accountName, storageEndpointSuffix, srcContainerName, accountSasToken)
+			dstPath := fmt.Sprintf("https://%s.blob.%s/%s%s", accountName, storageEndpointSuffix, dstContainerName, accountSasToken)
+
+			klog.V(2).Infof("copy blob container %s to %s", srcContainerName, dstContainerName)
+			out, copyErr = exec.Command("azcopy", "copy", srcPath, dstPath, "--recursive", "--check-length=false").CombinedOutput()
+			if copyErr != nil {
+				if strings.Contains(string(out), azcopyAuthenticationFailedCode) {
+					klog.Warningf("CopyContainer(%s, %s, %s) failed with error(%v): %v, sleep 2 min and retry", resourceGroupName, accountName, dstContainerName, err, string(out))
+					time.Sleep(2 * time.Minute)
```

Comment:
try tune this value a little bit and also it's better to have a wait for succeed process in the CreateVolume func, and we should enlarge `timeout` as 10min:
```
        - name: csi-provisioner
          image: mcr.microsoft.com/oss/kubernetes-csi/csi-provisioner:v3.5.0
          args:
            - "-v=2"
            ...
            - "--timeout=120s"
 ```